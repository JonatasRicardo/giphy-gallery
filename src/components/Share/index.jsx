
import React, { useEffect, useState } from 'react';
import { useTheme } from "react-jss";
import PropTypes from 'prop-types';
import useStyles from "./share.styles";

const propTypes = {
  url: PropTypes.string
};

const copyToClipboard = str => {
  const el = document.createElement('textarea');
  el.value = str;
  el.setAttribute('readonly', '');
  el.style.position = 'absolute';
  el.style.left = '-9999px';
  document.body.appendChild(el);
  el.select();
  document.execCommand('copy');
  document.body.removeChild(el);
};

const Share = ({
  url
}) => {
    const [copyFeedback, setCopyFeedback] = useState(false);
    const theme = useTheme();
    const classes = useStyles({ theme });

    const copy = () => {
      copyToClipboard(url)
      setCopyFeedback(true);
    }

    useEffect(() => {
      if (copyFeedback) {
        setTimeout(() => setCopyFeedback(false), 1000);
      }
    }, [copyFeedback]);

    return (
      <div className={classes.share}>
        <button className={classes.copyButton} onClick={copy} disabled={copyFeedback}>
          {copyFeedback ? 'Link Copied' : 'Copy Link'}
        </button>
      </div>
    );
}

Share.propTypes = propTypes;

export default Share;
