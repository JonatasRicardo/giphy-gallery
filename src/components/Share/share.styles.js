import {
  createUseStyles
} from "react-jss";

const useStyles = createUseStyles({
  share: {
    display: 'flex',
    justifyContent: 'center'
  },
  copyButton: {
    background: ({
      theme
    }) => theme.dark.primary,
    color: ({
      theme
    }) => theme.dark.background,
    border: 0,
    padding: ({
      theme
    }) => theme.spacing,
    outline: 'none',
    cursor: 'pointer'
  }
});

export default useStyles;
