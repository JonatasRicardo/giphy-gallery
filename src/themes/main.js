const theme = {
  dark: {
    background: "#000",
    primary: "#fff"
  },
  light: {
    background: "#fff",
    primary: "#000"
  },
  spacing: 10
};

export default theme;
