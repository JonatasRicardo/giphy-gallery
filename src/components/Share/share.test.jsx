import React from 'react';
import { ThemeProvider } from "react-jss";
import {
  render, fireEvent, waitFor, screen
} from '@testing-library/react';
import Share from './index';
import themes from '../../themes/main';

const renderComponent = ({ theme }) =>
  render(
    <ThemeProvider theme={theme}>
      <Share />
    </ThemeProvider>
  );

test('should match snapshot', async () => {
  const { asFragment } = renderComponent({ theme: themes });
  expect(asFragment()).toMatchSnapshot()
});

test('should show the feedback', async () => {
  renderComponent({ theme: themes });
  expect(screen.getByTestId('copy-button')).toHaveTextContent('Copy Link');
  fireEvent.click(screen.getByTestId('copy-button'));
  expect(screen.getByTestId('copy-button')).toHaveTextContent('Link Copied');
});
