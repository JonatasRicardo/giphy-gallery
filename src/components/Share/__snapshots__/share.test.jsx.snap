// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="share-0-2-1"
  >
    <button
      class="copyButton-0-2-2 copyButton-d0-0-2-3"
      data-testid="copy-button"
    >
      Copy Link
    </button>
  </div>
</DocumentFragment>
`;
